core/drivers/tpm2.c

```diff
@@ -0,0 +1,391 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Linaro Limited
+ *
+ */
+
+#include <kernel/delay.h>
+#include <tpm2.h>
+#include <trace.h>
+
+static bool tpm2_check_ops(struct tpm2_ops *ops)
+{
+	if (!ops || !ops->rx32 || !ops->tx32 || !ops->rx8 || !ops->tx8)
+		return false;
+
+	return true;
+}
+
+static bool tpm2_check_locality(struct tpm2_chip *chip, int loc)
+{
+	struct tpm2_ops *ops = chip->ops;
+	uint8_t locality = 0;
+
+	ops->rx8(chip, TPM2_ACCESS(loc), 1, &locality);
```

Comment:
It would be nice to have helper functions like so this call implement `tpm2_rx8(chip, args....);`